calcom/cal.com
# Implement Twilio for booking reminders

## What does this PR do?

This PR uses Twilio to send booking reminders to attendees. The reminders are set by the organizer. 

Addresses #1938 & #1780 

## Type of change




## How should this be tested?



### Testing SMS Reminders
1. Signup for a Twilio account
2. Create a number (Should be able to do so on homepage)
3. Add the the account id and auth token to .env
4. Create a messaging service and add the created number
5. Add the messaging_sid to the .env
6. Create a new event type, in the advanced options add an SMS reminder
7. Create a new booking, note that the scheduled send date should be within 60mins of the current time and <7 days
8. Go to the Twilio portal and go to messaging overview, you should see the message scheduled
	- Optionally you can also comment out the sendAt parameter to get the text message immediately 
9. Reschedule the booking, in the DB under attendeeReminder you should see the updated scheduled date
10. Cancel the booking, you should see the reminder is no longer in the DB and on the Twilio console the message should say cancelled

### Testing Email reminders
1. Signup for a Sendgrid account
2. Get an API key and setup the approved sender email
3. Add the API key and sender email to .env
4. Create a new event type, in the advanced options add an email reminder
5. Schedule a booking under that event type that is within 60 mins
6. You should receive a reminder email
7. Schedule another booking that is 1 day in advance.
    - If you do not receive an error booking the event then the email is scheduled
